### ISSUE TITLE
Feature request: Case sensitivity and case conversion

### ISSUE THREAD

**Issue by cheny0y0** [LLK/scratch-gui#7619]
### 1. Case sensitivity
The block ![(Hello)=(Hello)](https://user-images.githubusercontent.com/66211043/132947339-86f656e6-0d84-484e-a61e-ca5493bc8a79.jpeg) would be true, and the block 
![(Hello)=(hello)](https://user-images.githubusercontent.com/66211043/132947469-789ae1af-48a3-4892-8fdc-b04d7b8d9ce9.jpeg) would be false.
### 2. Case conversion
Request to add 2 blocks: one is to convert all the letters in the characters to uppercase (or lowercase); the other is to convert the first letter of each word to uppercase, and the rest to lowercase.

**Comment by ghost** [comment]
I request a seperate block for this purpose maybe a case sensitive equal to or a double equal to will be cool but don't make the existing block case sensitive. Also again who knows if this feature would be added or not. most probably not as kids would get confused.

**Comment by cheny0y0** [comment]
> I request a seperate block for this purpose maybe a case sensitive equal to or a double equal to will be cool but don't make the existing block case sensitive. Also again who knows if this feature would be added or not. most probably not as kids would get confused.

Do the kids think “Hello” equals to “hello”？ 😄

**Comment by BryceLTaylor** [comment]
It is intentional to have the equals operator in the core blocks be case INsensitive.  It makes it easier to use, especially for kids.

**Comment by Secret-chest** [comment]
> It is intentional to have the equals operator in the core blocks be case INsensitive.  It makes it easier to use, especially for kids.  

But then add a way to make them case-sensitive! @BryceLTaylor

**Comment by Quantum-Codes** [comment]
> > It is intentional to have the equals operator in the core blocks be case INsensitive.  It makes it easier to use, especially for kids.
> 
> But then add a way to make them case-sensitive! @BryceLTaylor

It will be hard for a beginner not knowing to code to deal with case sensitive things.

If anyone really wants it:(this is a workaround)
* Costume names are actually case sensitive. 
* Make a sprite with costumes named a-z and A-Z (26×2 costumes)
* Encode each letter with the costume #. You will get a series of numbers such as `128382029`. 
* Add an "a" in front to make equality checks in string and not integer (integer equality breaks at 15 digits I guess). Example: `a128382029`
* check whether the numbers are equal.
Basically, encode by using any cloud encoder and check whether they are equal or not
